### ISSUE TITLE
Named parameter has no value in the given object.

### ISSUE THREAD

**Issue by ben-eb** [sequelize/sequelize#9440]
## What are you doing?

I'd like to use named parameters - query below is contrived, I would expect to send more parameters than this when doing a real query. Unfortunately the regex for finding replacements does not take into account values inside string literals.

```js
const Sequelize = require('sequelize');

const sequelize = new Sequelize(process.env.DATABASE_URI, {
  dialect: 'postgres',
});

const query = `
  SELECT * FROM "_Join:roles:_Role" WHERE "owningId" = :owningId
`;

sequelize.query(query, {
  type: sequelize.QueryTypes.SELECT,
  replacements: {
    owningId: 'TQdJ6RuES6'
  }
}).then(console.warn)
```

Relevant line is here:

https://github.com/sequelize/sequelize/blob/v4/lib/sql-string.js#L112

## What do you expect to happen?

Sequelize should not try to find a named parameter inside a string literal.

## What is actually happening?

Sequelize is trying to find a `:roles` parameter.

```
Unhandled rejection Error: Named parameter ":roles" has no value in the given object.
    at sql.replace (/app/node_modules/sequelize/lib/sql-string.js:120:13)
    at String.replace (<anonymous>)
    at Object.formatNamedParameters (/app/node_modules/sequelize/lib/sql-string.js:112:14)
    at Object.formatNamedParameters (/app/node_modules/sequelize/lib/utils.js:126:20)
    at Promise.try (/app/node_modules/sequelize/lib/sequelize.js:500:23)
    at tryCatcher (/app/node_modules/bluebird/js/release/util.js:16:23)
    at Function.Promise.attempt.Promise.try (/app/node_modules/bluebird/js/release/method.js:39:29)
    at retryParameters (/app/node_modules/sequelize/lib/sequelize.js:454:48)
    at /app/node_modules/retry-as-promised/index.js:39:21
    at Promise._execute (/app/node_modules/bluebird/js/release/debuggability.js:303:9)
    at Promise._resolveFromExecutor (/app/node_modules/bluebird/js/release/promise.js:483:18)
    at new Promise (/app/node_modules/bluebird/js/release/promise.js:79:10)
    at retryAsPromised (/app/node_modules/retry-as-promised/index.js:29:10)
    at Sequelize.query (/app/node_modules/sequelize/lib/sequelize.js:454:12)
    at Object.<anonymous> (/app/test.js:11:11)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:188:16)
```

__Dialect:__ postgres
__Database version:__ 9.6
__Sequelize version:__ 4.37.7
__Tested with latest release:__ Yes

**Comment by luanxuechao** [comment]
@sushantdhiman  is there any solutions

**Comment by ephys** [comment]
I believe this was fixed by #14472
